### PR TITLE
slof_balloon: Add slof_balloon case for SLOF

### PR DESCRIPTION
**Pull request by YongxueHong**
slof_balloon case is that verify SLOF info with balloon.

ID: 1645063
Signed-off-by: Yongxue Hong <yhong@redhat.com>

### CODE REVIEW

**Comment by YongxueHong** [comment]
Depends on #1531

File: qemu/tests/cfg/slof_balloon.cfg [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    login_timeout = 240
+    start_vm = no
+    pci_controllers = ''
+    balloon = balloon0
+    balloon_dev_devid = balloon0
+    balloon_dev_add_bus = yes
+    only ppc64le ppc64
+    variants:
+        - with_pci_bridge:
+            pci_controllers +=" pci_bridge0"
```
Comment (ybduan) [inline qemu/tests/cfg/slof_balloon.cfg]: I think Line 8 can be deleted and Line 15 can be changed to 'pci_controllers = pci_bridge0'.

File: qemu/tests/slof_balloon.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            return True
+        else:
+            time.sleep(step)
+    return False
+
+
+@error_context.context_aware
+def run(test, params, env):
+    params['start_vm'] = 'yes'
+    env_process.preprocess_vm(test, params, env, params["main_vm"])
```
Comment (ybduan) [inline qemu/tests/slof_balloon.py]: Line 30 and Line 31 are redundant.

Comment (YongxueHong) [reply qemu/tests/slof_balloon.py]: Thanks your reviewing , will update it.

File: qemu/tests/slof_balloon.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    if int(balloon_size) != (int(params["mem"]) * 1024 * 1024):
+        test.error('The balloon size is not equal to %d'
+                   % (int(params["mem"]) * 1024 * 1024))
+
+    error_context.context('Change the balloon size to %s'
+                          % params['balloon_size'])
+    qmp_port.balloon(int(params['balloon_size']))
+    error_context.context('Check balloon size after changed.')
+    if not _wait_for_balloon_work(int(params['balloon_size']), qmp_port,
+                                  int(params['balloon_timeout'])):
```
Comment (wangxianxian) [inline qemu/tests/slof_balloon.py]: For 'balloon_timeout', the time that 'balloon' need will different for different balloon value, i.e, maybe timeout=300 is not ok for 'balloon value=200G',  so, it is better to specify a fixed value for "balloon value" for example 2g.

**Comment by wangxianxian** [comment]
Host_RHEL.RHEL_ALT.m7.u6.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.ALT.7.6.ppc64le.io-github-autotest-qemu.slof_balloon.with_pci_bridge: PASS

File: qemu/tests/slof_balloon.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        if not qmp_ports:
+            test.error("Incorrect configuration, no QMP monitor found.")
+        return qmp_ports[0]
+
+    def _check_balloon_info():
+        """ Check virtio balloon device info. """
+        error_context.context('Check virtio balloon device info.')
+        balloon_size = qmp.query('balloon')['actual']
+        logging.debug('The balloon size is %s' % balloon_size)
+        mem = int(params["mem"])
```
Comment (hereischen) [inline qemu/tests/slof_balloon.py]:
```python
 mem = int(params["mem"]) * 1024 * *2:
```

Comment (YongxueHong) [reply qemu/tests/slof_balloon.py]: Thanks, have updated.

**Comment by hereischen** [comment]
Acked-by: Haotong Chen hachen@redhat.com

**Comment by wangxianxian** [comment]
ACK

**Comment by YongxueHong** [comment]
Thanks all

**Pull request merged**